__tests__/functional/register.test.js

```diff
@@ -0,0 +1,125 @@
+import AppRunner from "./utils/AppRunner.js";
+import {
+    expectNoErrors,
+    expectAnyError,
+    expectAnyErrorMessageToBe,
+} from "./utils/matchers.js";
+
+const collectionsEmptiedOnCleanup = ["users"];
+let SUT = new AppRunner(collectionsEmptiedOnCleanup);
+
+const registerQuery = `mutation registerUser($username: String!, $password: String!, $email: String!) {
+							register(username: $username, password: $password, email: $email) {
+								user_id
+								username
+							}
+						}`;
+
+function createRegisterUserMsg() {
+    return {
+        query: registerQuery,
+        operationName: "registerUser",
+        variables: {
+            username: "test_user",
+            email: "grzesiekkasprzak", //currently this email is accepted!
+            password: "qwerty",
+        },
+    };
+}
+
+describe("register users", () => {
+    const registerUserMsg = createRegisterUserMsg();
+
+    beforeAll(async () => {
+        await SUT.start();
+    });
+
+    beforeEach(async () => {
+        expect(await SUT.getDocumentsCount("users")).toBe(0);
+    });
+
+    afterEach(async () => {
+        await SUT.cleanup();
+    });
+
+    afterAll(async () => {
+        await SUT.stop();
+    });
+
+    async function expectSuccessfulRegistration(request) {
+        const userCount = await SUT.getDocumentsCount("users");
+
+        var response = await SUT.request()
+            .post("/graphql")
+            .send(request)
+            .expect(200);
+
+        expectNoErrors(response);
+        expect(response.body.data.register.username).toBe(
+            request.variables.username
+        );
+        expect(await SUT.getDocumentsCount("users")).toBe(userCount + 1);
+
+        return response;
+    }
+
+    test("register a user, should respond with registered username and add user to database", async () => {
+        await expectSuccessfulRegistration(registerUserMsg);
+    });
+
+    //currently this password is accepted and test fails!
+    test.skip("register a user with empty password, should return error", async () => {
+        let registerUserEmptyPassword = createRegisterUserMsg();
+        registerUserEmptyPassword.variables.password = "";
+
+        const response = await SUT.request()
+            .post("/graphql")
+            .send(registerUserEmptyPassword)
+            .expect(200);
+
+        expectAnyError(response);
+        expect(await SUT.getDocumentsCount("users")).toBe(1);
+    });
```

Comment:
Same problem as above. This test should throw an error and not be skipped.

Reply:
Test throws an error and that's why it's skipped.
This review request delivers testing setup and these few tests are only examples of what we can do now. Accidentally one of them revealed a bug. I notified you about it and I assume it's gonna be fixed (under **new** task on trello).
This 'skip' should be removed no sooner than the application is fixed and test passes.

Reply:
I created a card on Trello (no 64) for adding password validation on backend (effectively making this test pass) and removing .skip.